globalstate.js

```diff
@@ -29,6 +29,44 @@ function getAlpineState() {
       return phoneRegex.test(phone.trim().replace(/\s/g, ''));
     },
 
+    validateYtunnus(ytunnus) {
+      if (!ytunnus || !ytunnus.trim()) {
+        return false;
+      }
+
+      const trimmedYtunnus = ytunnus.trim();
+
+      // Check format: XXXXXXX-Y (7 digits, dash, 1 digit)
+      const ytunnusRegex = /^\d{7}-\d{1}$/;
+      if (!ytunnusRegex.test(trimmedYtunnus)) {
+        return false;
+      }
+
+      // Extract digits and check digit
+      const digits = trimmedYtunnus.replace('-', '');
+      const checkDigit = parseInt(digits.charAt(7));
+
+      // Weights for Y-tunnus validation
+      const weights = [7, 9, 10, 5, 8, 4, 2];
+      let sum = 0;
+
+      // Calculate weighted sum
+      for (let i = 0; i < 7; i++) {
+        sum += parseInt(digits.charAt(i)) * weights[i];
+      }
+
+      // Calculate check digit
+      let calculatedCheckDigit = (11 - (sum % 11)) % 11;
+
+      // If result is 11, check digit becomes 0
+      if (calculatedCheckDigit === 11) {
+        calculatedCheckDigit = 0;
+      }
```

Comment:
Lines 61-64 can be removed, because you already do `% 11` to the result on line 59 (so the condition `calculatedCheckDigit === 11` is no longer possible).